src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -18,12 +25,64 @@ private Drivetrain() {
      * @param rightSpeed Speed of right side.
      */
     public void tankDrive(double leftSpeed, double rightSpeed) {
+
+        if (!checkDriveMotorsGoCorrectDirection(leftSpeed, rightSpeed)) {
+            this.stop();
+        }
     }
 
-    public static Drivetrain getInstance() {
-        return INSTANCE;
+    /**
+     * checkDriveMotorsGoCorrectDirection sanity check
+     * this checks to make sure that if the robot is told to go forward, its actually traveling forward instead of backward
+     * @param leftSpeed  Speed of left side.
+     * @param rightSpeed Speed of right side.
+     * leftCounter uses Math.signum to get the signs of the three left motors.
+     * if the motor is traveling forward, leftCounter increments by 1. If backward, it decrements by 1.
+     * leftCounter  will have values of 3 or -3 if all motors travel in the same direction.
+     * the if statement makes sure all motors are going the same direction,
+     *                   and that they are traveling the same direction as the variable leftSpeed is telling them to go
+     * Ditto for right side.
+     */
+    private boolean checkDriveMotorsGoCorrectDirection(double leftSpeed, double rightSpeed) {
+        int leftCounter = 0;
+        leftCounter += Math.signum(PDP.getInstance().getCurrent(RobotMap.DRIVETRAIN.leftMasterChannel));
+        leftCounter += Math.signum(PDP.getInstance().getCurrent(RobotMap.DRIVETRAIN.leftFollower1Channel));
+        leftCounter += Math.signum(PDP.getInstance().getCurrent(RobotMap.DRIVETRAIN.leftFollower2Channel));
+
+        if (Math.abs(leftCounter) != 3 || Math.signum(PDP.getInstance().getCurrent(RobotMap.DRIVETRAIN.leftMasterChannel)) != Math.signum(leftSpeed)) {
```

Comment:
So I did sleep on it, and I'm still looking at this thinking that there might be a situation where the motors are drawing "negative current" according the the PDP when a positive signal is sent, but that remains desired operation. I'm going to have to test this on a practice base though. If that is the case, we have to do some changes to this statement (because I doubt @Feith27 will be likely to hardware invert motors.)

If nothing else, this is definitely something that I want to see "battle tested" on real robots.